server/broadcast.go

```diff
@@ -56,34 +56,98 @@ var submitMultiSession = func(ctx context.Context, sess *BroadcastSession, seg *
 var maxTranscodeAttempts = errors.New("hit max transcode attempts")
 
 type BroadcastConfig struct {
-	maxPrice *core.AutoConvertedPrice
-	mu       sync.RWMutex
+	maxPricePerCapability map[core.Capability]map[string]*core.AutoConvertedPrice
+	mu                    sync.RWMutex
 }
 
 type SegFlightMetadata struct {
 	startTime time.Time
 	segDur    time.Duration
 }
 
+func (cfg *BroadcastConfig) Initialize() {
+	//initialize with base capability of H264 transcoding
+	cfg.maxPricePerCapability = make(map[core.Capability]map[string]*core.AutoConvertedPrice)
+	models := make(map[string]*core.AutoConvertedPrice)
+	models["default"] = core.NewFixedPrice(big.NewRat(0, 1))
+	cfg.maxPricePerCapability[core.Capability_H264] = models
+}
+
 func (cfg *BroadcastConfig) MaxPrice() *big.Rat {
 	cfg.mu.RLock()
 	defer cfg.mu.RUnlock()
-	if cfg.maxPrice == nil {
+	//base price is H264 encoding
+	if cfg.maxPricePerCapability[core.Capability_H264]["default"] == nil {
 		return nil
 	}
-	return cfg.maxPrice.Value()
+	return cfg.maxPricePerCapability[core.Capability_H264]["default"].Value()
 }
 
 func (cfg *BroadcastConfig) SetMaxPrice(price *core.AutoConvertedPrice) {
 	cfg.mu.Lock()
 	defer cfg.mu.Unlock()
-	prevPrice := cfg.maxPrice
-	cfg.maxPrice = price
+	prevPrice := cfg.maxPricePerCapability[core.Capability_H264]["default"]
+	cfg.maxPricePerCapability[core.Capability_H264]["default"] = price
 	if prevPrice != nil {
 		prevPrice.Stop()
 	}
 }
 
+// gets prices for a set of capabilities
+func (cfg *BroadcastConfig) GetCapabilitiesMaxPrice(caps common.CapabilityComparator) *big.Rat {
+	cfg.mu.RLock()
+	defer cfg.mu.RUnlock()
+	netCaps := caps.ToNetCapabilities()
+	price := big.NewRat(0, 1)
+	for capabilityInt, constraints := range netCaps.Constraints.PerCapability {
+		for modelID, _ := range constraints.Models {
+			capPrice := cfg.GetCapabilityMaxPrice(core.Capability(capabilityInt), modelID)
+			if capPrice != nil {
+				price = price.Add(price, capPrice)
+			}
+		}
+	}
+
+	//if no prices set per model, return maxPrice
+	if price.Cmp(big.NewRat(0, 1)) == 0 {
+		price = cfg.MaxPrice()
+	}
+
+	return price
+}
+
+func (cfg *BroadcastConfig) GetCapabilityMaxPrice(cap core.Capability, modelID string) *big.Rat {
```

Comment:
This one can be lowercase since it is not used outside this file :).

Reply:
Updated